storage/src/main/java/tech/pegasys/artemis/storage/utils/UnsignedLongSerializer.java

```diff
@@ -15,21 +15,38 @@
 
 import com.google.common.primitives.UnsignedLong;
 import java.io.IOException;
-import java.io.Serializable;
+import java.util.Comparator;
+import org.jetbrains.annotations.NotNull;
 import org.mapdb.DataInput2;
 import org.mapdb.DataOutput2;
-import org.mapdb.Serializer;
+import org.mapdb.serializer.SerializerEightByte;
 
-public class UnsignedLongSerializer implements Serializer<UnsignedLong>, Serializable {
+public class UnsignedLongSerializer extends SerializerEightByte<UnsignedLong> {
 
   @Override
-  public void serialize(DataOutput2 out, UnsignedLong value) throws IOException {
-    out.writeChars(value.toString(10));
+  protected UnsignedLong unpack(final long data) {
+    return UnsignedLong.fromLongBits(data);
   }
 
   @Override
-  public UnsignedLong deserialize(DataInput2 in, int available) throws IOException {
-    UnsignedLong returnVal = UnsignedLong.valueOf(in.readLine(), 10);
-    return returnVal;
+  protected long pack(final UnsignedLong value) {
+    return value.longValue();
+  }
```

Comment:
Why did we need to implement these pack and value array search methods?

Reply:
They're required to support using that type in sorted collections and we want to be able to store the slot -> root lookup in order to make it efficient to search backwards.